packages/replicache/src/replicache-options.ts

```diff
@@ -4,12 +4,19 @@ import type {StoreProvider} from './kv/store.ts';
 import type {Puller} from './puller.ts';
 import type {Pusher} from './pusher.ts';
 import type {MutatorDefs, RequestOptions} from './types.ts';
+import type {Hash} from './hash.ts';
+import type {MaybePromise} from '../../shared/src/types.ts';
+import type {InternalDiff} from './btree/node.ts';
+import type {Store} from './dag/store.ts';
 
 /**
  * The options passed to {@link Replicache}.
  */
 
-export interface ReplicacheOptions<MD extends MutatorDefs> {
+export interface ReplicacheOptions<
+  MD extends MutatorDefs,
+  TZeroData extends ZeroTxData = ZeroTxData,
```

Comment:
I think it would be better to create a new interface that extends ReplicacheOptions. ReplicacheImpl can use that type. That way we do not have to change the public API of replicache.

Reply:
I think you can either add one more option here:

https://github.com/rocicorp/mono/blob/de356bdedc85e76a6622d7c110f5ed5d8e152402/packages/replicache/src/replicache-impl.ts#L388-L391

or you can extend `ReplicacheImplOptions` which is all internal stuff the Zero and Reflect use.